### ISSUE TITLE
Add per game settings as an option on Android.

### ISSUE THREAD

**Issue by RichieMan007** [Lime3DS/Lime-3DS-Emulator#38]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

If possible can per game settings be added, so users can adjust settings for each individual game.

### Why would this feature be useful?

As you may need to adjust resolution or use a different graphics driver to get a game performing better.

**Comment by RichieMan007** [comment]
Thanks for adding my request to the Todo list.

**Comment by nevmind** [comment]
oh great. Per-Game settings is very useful feature for games which needs to be set specific region.

**Comment by FrogProjectOfficial** [comment]
> oh great. Per-Game settings is very useful feature for games which needs to be set specific region.

Also, some games work better with OpenGLES than Vulkan - would be great to have per-game settings for situations like this, especially when using with front-ends.

**Comment by AmandaRoseChaqueta** [comment]
Isn't this already a thing?

![Image](https://github.com/user-attachments/assets/f70a438a-09ad-42e2-8eff-1359bc93f6bc)

![Image](https://github.com/user-attachments/assets/78731234-b969-42a3-a11c-a99bc0dabbab)

**Comment by palop4l** [comment]
> Isn't this already a thing?
> 
> ![Image](https://github.com/user-attachments/assets/f70a438a-09ad-42e2-8eff-1359bc93f6bc)
> 
> ![Image](https://github.com/user-attachments/assets/78731234-b969-42a3-a11c-a99bc0dabbab)

Not in mobile/android version

**Comment by palop4l** [comment]
How do you bump up issue? Anyway, yes please i would like this to be prioritized even higher if possible. Add per-game custom layout and ability to set more than one custom layout per-game while we're at it.

**Comment by Kazoid1337** [comment]
I second this, this is vital to make any games on android playable. Every game needs it's graphics settings tweaked, how can anyone play multiple games if they have to set (and remember) their graphics settings and change them each and every single time? This needs to be the number 1 priority right now.

**Comment by RenanSD007** [comment]
This would be good especially for custom layouts, something like melonDS Android

**Comment by leira** [comment]
It's great this feature is already on desktop version. It will be great to see this feature ported to the Android version. Can you please change the title of this issue. to reflect it?

**Comment by hiresgaming89** [comment]
Please if possible make it high priority because many dual display android handheld are lunching and games need specific settings like resolution,  controls, vulkan/opengl, etc etc. Thank you for all your works.

**Comment by EthanDonley** [comment]
Bumping up this issue, I would really like to see per-game configs!

**Comment by Nexu36** [comment]
I'd also like to see this implemented, and it would be quite a big quality of life improvement.

Some games won't run well with certain configurations (e.g. requiring opengl over vulkan) or require a lowering of resolution. This can get difficult to keep track of manually, and it's somewhat of the norm for android emulators to have per-game configurations so it would be great to see this implemented.

**Comment by jettsd** [comment]
bumping

**Comment by slowthisbirddown** [comment]
please

**Comment by Nexu36** [comment]
I can see someone's submitted a PR for per-game config via ini settings (#1401). It's not entirely clear how one would go about editing these config files but this looks essentially like it would satisfy this issue.

**Comment by SethG911** [comment]
Bumping. I need to be able to use LLE modules to use Pretendo for specific games and not others, or they lock up.

**Comment by pakrett** [comment]
Can someone please merge this commit ([#1401](https://github.com/azahar-emu/azahar/pull/1401)) this is a feature that WE NEED on android, every mature emulator should have this. I'm currently using azahar + along with azahar just because like that I can have at least two different configs this is ridiculous.

**Comment by DavidRGriswold** [comment]
When I last tested this it did not work correctly, and I have not heard that it was fixed. Feel free to test the CI build and give feedback on whether it works or not.

**Comment by pakrett** [comment]
> When I last tested this it did not work correctly, and I have not heard that it was fixed. Feel free to test the CI build and give feedback on whether it works or not.

Where can I find it?

**Comment by DavidRGriswold** [comment]
on the PR you linked, there should be a tab at the top called Checks, one of which is the build for the android version of that PR. Download it on your device, install, test, let us know.

**Comment by pakrett** [comment]
You are right, this does not work correctly and need some fixes. When I have tested it, I was able to create a custom settings for my game but the parameters I change didn't remain changed. Even if I manually edit the custom setting ".ini" to apply my changes and this time the interface tells me that these parameters are indeed activated the game doesn't take this into account and loads the main parameters.

**Comment by ShazamaPajama** [comment]
I would very much appreciate this feature for Azahar on Android. Please. Pretty Please?

**Comment by DavidRGriswold** [comment]
If you are willing to test, a draft PR of this feature exists at #1893 . It needs significant testing as there are a *lot* of settings to test, and this took significant changes to get working correctly.

**Comment by Hyzian** [comment]
This is a vital feature that every single emulator should have, PLEASE make it high priority!

**Comment by lipowskm** [comment]
> This is a vital feature that every single emulator should have, PLEASE make it high priority!

Bro is too busy fighting piracy it seems, instead of pushing this, let's be honest, easy to implement and basically vital feature.

**Comment by DavidRGriswold** [comment]
> > This is a vital feature that every single emulator should have, PLEASE make it high priority!
> 
> Bro is too busy fighting piracy it seems, instead of pushing this, let's be honest, easy to implement and basically vital feature.

If it's easy to implement why haven't you done it?

**Comment by playnitoxp** [comment]
Some of the games I play require different settings to work properly, I request the Devs to please bump up the priority to high many people like me are waiting for this feature to be implemented, now that we have our Thors in our hands.

**Comment by Sucophobia** [comment]
bumping this, very vital feature for any sort of per game tinkering

**Comment by KizGalaxy** [comment]
Bump! Different games require different settings and having to change them every time I want to play is very tedious.

**Comment by manutortosa-collab** [comment]
At this stage of development and with the different Turnip drivers and the great success of the Android dual screen devices as the Ayn Thor, you could consider start adding even some initial implementation as per game settings are almost a must in Android.

**Comment by Cyber-Tokyo** [comment]
Bumping this. Per game settings would be awesome.

**Comment by sks316** [comment]
Absolutely wild to me that this isn't implemented yet. I had some graphics issues with Mario & Luigi Dream Team on Vulkan and it would be nice to have a per-game setting that switches it to OpenGLES automatically.

**Comment by kay069** [comment]
Still not implemented yet 😭😭😭😭😭😭

**Comment by IISAKNIFE** [comment]
Any update on when this could be implemented?

**Comment by vl3214** [comment]
Pablo is a complete jackass, he's so focused on marking other requests for this feature as duplicates instead of actually implementing the feature, meanwhile every emulator has had this shit for years. I'm hoping the AzaharPlus fork will get this feature in the future, as it's obvious the extremely poorly developed Azahar will never get this feature. AzaharPlus is also just superior anyways, no constant game crashing like this has, I suggest everyone switch to it and stop supporting this slop.

Lets be real, Azahar is mostly just copy pasted Citra with little to no meaningful changes happening. This is what happens when script kiddies take over where the actual devs left off, unfortunately.

**Comment by DavidRGriswold** [comment]
"AzaharPlus" literally states in their changelog that their fork is exactly azahar with very few changes. If you get a crash in azahar, please report a bug report so it can get looked at

As to this request, it is already in the works and should be available within the next few releases, as already mentioned in several comments here. Please read before going on a rant. Cleaning up GitHub issues is a requirement for managing a project this large.

**Comment by ThunderDragonRandy** [comment]
> Pablo is a complete jackass, he's so focused on marking other requests for this feature as duplicates instead of actually implementing the feature, meanwhile every emulator has had this shit for years. I'm hoping the AzaharPlus fork will get this feature in the future, as it's obvious the extremely poorly developed Azahar will never get this feature. AzaharPlus is also just superior anyways, no constant game crashing like this has, I suggest everyone switch to it and stop supporting this slop.
> 
> Lets be real, Azahar is mostly just copy pasted Citra with little to no meaningful changes happening. This is what happens when script kiddies take over where the actual devs left off, unfortunately.

[This feature](https://github.com/azahar-emu/azahar/pull/1856) is mentioned in the [2127.0 milestone](https://github.com/azahar-emu/azahar/milestone/33).
